spec/models/preference_spec.rb

```diff
@@ -19,11 +19,28 @@
 require 'rails_helper'
 
 RSpec.describe Preference do
-  describe 'validations' do
-    subject { build(:preference) }
+  subject { create(:preference, user:) }
```

Comment:
Aca seria mejor usar `build` en vez de `create` para no persistir el objeto al probar las validaciones.
Despues en el bloque `it` podrias hacer otro `build` y verificar si es `valid?` para que de el error o podrías ajustar el `subject` que tenes ahora a un `subject.save!` para que te funcione el `raise_error`. Creo que esas opciones deberían funcionar.

Reply:
Claro, entiendo. Al final usé el subject.save! porque creo que lo veo más explícito que va a tirar la excepción ahí